src/components/molecules/LogoutButton/LogoutButton.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState } from 'react';
+import { useGoogleLogout } from 'react-google-login';
+import { useAppDispatch } from 'src/config/store';
+import { setUser } from 'src/store/common/common.slice';
+import { MenuItem, Typography } from '@mui/material';
+import { useNavigate } from 'react-router-dom';
+import { AppRoutes } from 'src/config/constants';
+
+interface Props {
+  handleClose: () => void;
+  loggedInWith: string;
+}
+
+const LogoutButton: React.FC<Props> = ({ handleClose, loggedInWith }) => {
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
+
+  const onLogoutSuccess = () => {
+    handleClose();
+    dispatch(setUser(null));
+    localStorage.removeItem('userId');
+    navigate(AppRoutes.LOGIN);
+  };
+
+  const onFailure = () => {
+    console.log('Handle failure cases');
+  };
+
+  const { signOut } = useGoogleLogout({
+    clientId: process.env.REACT_APP_GOOGLE_ID as string,
+    onLogoutSuccess,
+    onFailure
+  });
+
+  const handleLogOut = () => {
+    switch (loggedInWith) {
+      case 'facebook':
+        onLogoutSuccess();
+        window.FB.api('/me/permissions', 'delete', null, () => window.FB.logout());
+        break;
+      case 'google':
+        signOut();
```

Comment:
I think it is better to name these functions as `googleSignOut` and `facebookSignOut` because `onLogoutSuccess` and just `signOut ` don't show assignment to login type

Reply:
agree)